ecommerce/OrderController.php

```diff
@@ -156,6 +163,48 @@ public function get_order_by_id() {
 		);
 	}
 
+	/**
+	 * Mark an order as paid.
+	 *
+	 * This function verifies a nonce for security, constructs a payload object with
+	 * the order ID, note, and payment status, and updates the payment status of the order
+	 * to 'paid'. It sends a JSON response indicating the success or failure of the operation.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return void
+	 */
+	public function order_mark_as_paid() {
+		if ( ! tutor_utils()->is_nonce_verified() ) {
```

Comment:
Please Add permission check after nonce validation like below:
`
if ( ! current_user_can( 'manage_options' ) ) {
    $this->json_response( MESSAGE, null, STATUS_CODE );
}
`